EIPS/eip-3712.md

```diff
@@ -0,0 +1,337 @@
+    
+    event TransferBatchNoIndexed(uint256[] ids,address[] from,address[] to,uint256[] value);
+    
+    event ApproveBatch(uint256[] ids,address indexed from,address indexed to,uint256[] value);
+    
+    event ApproveBatchNoIndexed(uint256[] ids,address[] from,address to,uint256[] value);
+}
+```
+
+Each of the interfaces `totalSupply`,`balanceOf`,`transfer`,`allowance`,`transferFrom`,`approve(uint256 id,address spender, uint256 amount)`  introduces a new parameter id  in the [ERC-20](./eip-20.md)'s corresponding interface's parameter list to specify a fungible token's type. 
+
+
+`approve(address spender,bool status)` is similar to [ERC-1155](./eip-1155.md)'s interface. It allows an approval to be set uniformly for all types of tokens.
+
+`transferBatch` and `transferFromBatch` remove the parameter bytes memory data that the [ERC-1155](./eip-1155.md) standard uses and implement batch transactions.
+
+`transferBatchMul` implements transactions of multiple types of fungible tokens to multiple receiving addresses.
+
+`transferFromBatchMul` implements transactions of multiple types of fungible tokens from multiple sending addresses to multiple receiving addresses.
+
+
+## Rationale
+
+The [ERC-20](./eip-20.md) token standard defines a single fungible token. The [ERC-1155](./eip-1155.md) token standard defines multiple types of tokens and specifies each token type by using a parameter `id`. By  using this parameter `id`, the [ERC-1155](./eip-1155.md) token standard greatly reduces the gas consumption in transactions of multiple types of tokens, compared with the [ERC-20](./eip-20.md) token standard. Therefore our proposed standard uses a parameter `id` as well to specify a token type in order to reduce the gas consumption for  transactions of multiple types of tokens in one call operation via `transferBatch` or `transferFromBatch`. In our proposed st andard, each type of fungible token is specified by an id, therefore we add an interface `approve(uint256 id,address spender, uint256 amount)` for applications to set an approval for the id specified type of fungible token. In order to set approvals for multiple types of fungible tokens in minimum operations we introduce a function `approve(address spender,bool _status)`  to set approvals uniformly for all types of fungible tokens in one call operation. To handle a transaction of multiple types of fungible tokens from one sending address to multiple receiving addresses we introduce an interface `transferBatchMul` . To handle a transaction of multiple types of fungible tokens from multiple sending addresses to multiple receiving addresses we introduce an interface `transferFromBatchMul` .     
+
+## Backwards Compatibility
+
+Our proposed standard strictly adheres to the EIP rules and specifications, therefore our standard is backwork compatible with existing EIPs. Part of the transfer function is compatible with [ERC-1155](./eip-1155.md) but not [ERC-20](./eip-20.md).
```

Comment:
```suggestion
This EIP is partially backwards compatible with [EIP-1155](./eip-1155.md), but is NOT compatible with [EIP-721](./eip-721.md) or [EIP-20](./eip-20.md).
```

I suggest adding that it is not compatible with EIP-721 because it is also a commonly-used token standard (despite being a subset of EIP-1155).